cwlogger.go

```diff
@@ -6,18 +6,50 @@ import (
 )
 
 type logger struct {
-	id         string
-	sourceName string
-	sourceID   string
+	id           string
+	sourceName   string
+	sourceID     string
+	debugEnabled bool
 }
 
 var cwLogger logger
 
-// Init initializes the logger with the request id and prefix
+// Init initializes the logger with the request id and prefix. Using this initialization method will disable debug logging.
 func Init(requestID, sourceName string) {
+	InitWithDebugLevel(requestID, sourceName, false)
+}
+
+// InitWithDebugLevel initializes the logger with the request id and prefix. The parameter debugEnabled specifies if logging of log level Debug will be enabled or not.
+func InitWithDebugLevel(requestID, sourceName string, debugEnabled bool) {
 	cwLogger = logger{
-		id:         requestID,
-		sourceName: sourceName,
+		id:           requestID,
+		sourceName:   sourceName,
+		debugEnabled: debugEnabled,
+	}
+}
+
+// DebugStringf debug log helper to use sprintf formatting.
+func DebugStringf(format string, args ...interface{}) {
+	if cwLogger.debugEnabled {
+		DebugString(fmt.Sprintf(format, args...))
+	}
+}
+
+// DebugString logs a string message with DEBUG level
+func DebugString(msg string) {
+	if cwLogger.debugEnabled {
+		Debug(&LogEntry{Message: msg})
+	}
+}
+
+// Debug logs a message with DEBUG level
+func Debug(msg ILogEntry) {
+	if cwLogger.debugEnabled {
+		msg.SetLogLevel("DEBUG")
+		msg.SetRequestID(cwLogger.id)
+		msg.SetEventTime(time.Now())
+		msg.SetSourceName(cwLogger.sourceName)
+		fmt.Println(msg.Stringify())
 	}
```

Comment:
if ! debugEnabled { return } instead to make the code cleaner

Reply:
done